packages/react-ui/src/app/connections/create-edit-connection-dialog.tsx

```diff
@@ -74,15 +88,34 @@ const CreateOrEditConnectionDialog = React.memo(
 
     const formSchema = Type.Composite([
       Type.Object({
-        request: UpsertAppConnectionRequestBody,
+        request: Type.Intersect([
+          UpsertAppConnectionRequestBody,
+          Type.Object({
+            name: Type.String({
+              pattern: '^[A-Za-z0-9_\\-@\\+\\.]*$',
+              errorMessage: t(
+                ' Name can only contain letters, numbers and underscores',
+              ),
+            }),
+          }),
+        ]),
       }),
       overrideSchema,
     ]);
 
+    useEffect(() => {
```

Comment:
I don't think you need useEffect, reconnectionConnection is property, so it will trigger render

Check this: https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state